File: src/main/java/net/guizhanss/fastmachines/items/machines/generic/BasicFastMachine.java
package net.guizhanss.fastmachines.items.machines.generic;

import javax.annotation.ParametersAreNonnullByDefault;

import org.bukkit.inventory.ItemStack;

import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItemStack;
import io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType;

/**
 * The basic fast machine.
 * The default energy per use is 8 and the default capacity is 1024.
 *
 * @author ybw0014
 */
public abstract class BasicFastMachine extends AbstractFastMachine {

    @ParametersAreNonnullByDefault
    protected BasicFastMachine(SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
        super(item, recipeType, recipe, 8, 1024);
    }
}

File: src/main/kotlin/net/guizhanss/fastmachines/implementation/items/machines/generic/BasicFastMachine.kt
package net.guizhanss.fastmachines.implementation.items.machines.generic

import io.github.thebusybiscuit.slimefun4.api.items.ItemGroup
import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItemStack
import io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType
import org.bukkit.inventory.ItemStack

/**
 * The basic machine level fast machine.
 */
abstract class BasicFastMachine(
    itemGroup: ItemGroup,
    itemStack: SlimefunItemStack,
    recipeType: RecipeType,
    recipe: Array<out ItemStack?>,
) : AbstractFastMachine(itemGroup, itemStack, recipeType, recipe, 1024, 8)
